protobuf/types/service.proto

```diff
@@ -177,7 +177,7 @@ message Service {
   }
 
   // Service's hash.
-  string hash = 10 [
+  bytes hash = 10 [
```

Comment:
why not using gogoproto custom type to have directly the hash in the generated struct?
This question apply to every hash.
```
   (gogoproto.customtype) = "github.com/mesg-foundation/engine/hash.Hash", 
```

Reply:
because this pr is only about changing the hash type. Adding custom type will be in next pr